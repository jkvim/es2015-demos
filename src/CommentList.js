import React, { Component } from 'react'
import PropTypes from 'prop-types';

import Comment from './Comment';

class CommentList extends Component {
    static propTypes = {
        comments: PropTypes.arrayOf(Object)
    }

    componentWillMount() {
        console.log('Son componentWillMount');
    }

    componentDidMount() {
        console.log('Son componentDidMount');
    }

    componentWillReceiveProps() {
        console.log('Son componentWillReceiveProps');
    }

    shouldComponentUpdate(nextProps, nextState) {
        console.log('Son shouldComponentUpdate');
        return true;
    }

    componentWillUpdate(nextProps, nextState) {
        console.log('Son componentWillUpdate');
    }

    componentDidUpdate(prevProps, prevState) {
        console.log('Son componentDidUpdate');
    }

    render() {
        console.log('Son render');
        const { comments } = this.props
        return (
            <div>
                {comments.map((comment, i) =>
                    <Comment key={i} {...comment} />
                )}
            </div>
        )
    }
}

export default CommentList;
