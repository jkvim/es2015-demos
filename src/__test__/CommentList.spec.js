import React from 'react';
import renderer from 'react-test-renderer';

import CommentList from '../CommentList';


describe('<CommentList /> Snapshot testing', () => {
    
    it('should render correctly', () => {
        // given
        const comments = [
            {
                username: 'Tony',
                content: 'I have a pen, I have an apple'
            },
            {
                username: 'John',
                content: 'I have a pen, I have a pineapple'
            }
        ]

        // when
        const tree = renderer
            .create(<CommentList comments={comments} />)
            .toJSON()

        // then
        expect(tree).toMatchSnapshot();
    });
    
});
