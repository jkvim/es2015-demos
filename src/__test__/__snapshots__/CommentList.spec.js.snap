// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentList /> Snapshot testing should render correctly 1`] = `
<div>
  <div
    className="comment"
  >
    <p
      className="username"
    >
      Tony
    </p>
    :
    <p
      className="content"
    >
      I have a pen, I have an apple
    </p>
  </div>
  <div
    className="comment"
  >
    <p
      className="username"
    >
      John
    </p>
    :
    <p
      className="content"
    >
      I have a pen, I have a pineapple
    </p>
  </div>
</div>
`;
